### DISCUSSION TITLE
"cy.get().then()" - where can I find the source code?

### DISCUSSION THREAD

**Discussion by PalZol999** [cypress-io/cypress#29505] [Questions and Help]
Dear helpers!
I'm currently learning Cypress as a Junior QA, for a better understanding I would like to take a look of how the "cy.get().then()" has been written.
Can you please indicate me the file in the git repo where I can find the source code?
Thank you in advance

**Comment by portyu9** [comment]
Hi @PalZol999 

There isn't one implementation specifically for `cy.get().then()` — those are two separate Cypress commands being chained.

For the current Cypress source:

- `cy.get()` is implemented in:
  `packages/driver/src/cy/commands/querying/querying.ts`

  Look for:

  ```ts
  Commands.addQuery('get', ...)
  ```

- `.then()` is implemented in:
  `packages/driver/src/cy/commands/connectors.ts`

  The main callback logic is in `thenFn`.

Looking at those two files, along with Cypress's command queue implementation, should give you a good picture of how the chain works.

Please mark this answer as accepted if it helped, thank you.
